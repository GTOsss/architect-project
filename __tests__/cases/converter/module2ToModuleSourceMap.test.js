const stringifyObject = require('stringify-object');
const { module2ToModule } = require('../../../src/utils/change/module2ToModule');
const { aliases, map } = require('./source_map/source-map-module2');

const createPrettyObj = (obj) => {
  return stringifyObject(obj, {
    indent: '  ',
    singleQuotes: true,
  });
};

describe('test module2ToModule', () => {
  test('simple', () => {
    const obj = module2ToModule({
      map,
      aliases: {},
    });

    expect(createPrettyObj(obj)).toMatchSnapshot();
  });

  test('simple with aliases', () => {
    const obj = module2ToModule({
      map,
      aliases,
    });

    expect(createPrettyObj(obj)).toMatchSnapshot();
  });
});
