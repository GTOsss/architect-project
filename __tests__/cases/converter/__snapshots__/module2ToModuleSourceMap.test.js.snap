// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test module2ToModule simple 1`] = `
"{
  map: {
    'src/icons': {
      icons: {
        template: 'i',
        variable: '4'
      }
    },
    'src/component': {
      button: {
        template: 'icons'
      }
    }
  },
  aliases: {}
}"
`;

exports[`test module2ToModule simple with aliases 1`] = `
"{
  map: {
    'src/icons': {
      icons: {
        template: 'icons',
        variable: '4'
      }
    },
    'src/component': {
      button: {
        template: 'icons'
      }
    }
  },
  aliases: {
    c: 'component',
    s: 'store',
    ix: 'index',
    i: 'icons'
  }
}"
`;
